### PR TITLE
fix(background): stop endless OCR retry, fail-tag after OCR_MAX_RETRIES attempts

### PR DESCRIPTION
**Pull request by lunetics**
## Problem

When OCR processing of a document picked up by the auto-OCR poll fails, `processAutoOcrTagDocuments` just logs and `continue`s — the auto OCR tag stays on the document, so it is retried **every poll cycle, forever**. For persistently failing documents (e.g. a corrupt PDF that can't be split, or a per-document provider error) this:

- re-bills paid OCR providers for the same hopeless document on every cycle, and
- keeps the whole background loop in exponential backoff (the per-document error is aggregated into the loop error), throttling processing of unrelated healthy documents for up to an hour.

The fail-tag mechanism introduced in #990 (`FAIL_TAG`, `recoverFromFailedUpdate`) only covers failures of the *update* step after processing — failures of the OCR step itself had no loop-break at all.

*Related to #170 — but deliberately **not** claiming to fix it: #170 is primarily about the vision-LLM/Ollama call hanging and generating forever, which needs a timeout and is out of scope here. This PR only stops the cross-cycle retry loop for OCR attempts that return an error.*

## Fix

- **`OCR_MAX_RETRIES`** (default `3`, `0` restores the old retry-forever behavior): per-document count of failed OCR attempts. When the limit is reached, the auto OCR tag is removed and `FAIL_TAG` applied — same loop-break + review-marker semantics that update failures already have (those give up after a *single* attempt, so 3 is the more lenient setting).
- The tag-surgery core of `recoverFromFailedUpdate` is extracted into a generic `markProcessingFailed(...) error` helper (the old function stays as a thin wrapper with unchanged behavior). The new call site checks the error: if the recovery PATCH itself fails, the counter is kept so recovery is retried on the next cycle.
- The counter lives in memory on `App` (mutex-protected). It is reset as soon as OCR succeeds for the document.
- A shutdown of the poll loop is never counted (checked via the loop's own `ctx`, not by matching `context.Canceled` in the returned error — providers like Azure DocAI wrap their calls in their own timeout contexts, and those per-document timeouts *do* count toward the limit).
- Once a document has been fail-tagged, its (handled) error is no longer fed into the aggregated loop error, so an already-quarantined document can't keep the loop in backoff.

## Trade-off: transient provider outages

With the loop's exponential backoff, three failed attempts can happen within roughly a minute — so a short provider outage can fail-tag pending documents, which then need manual re-tagging. That is a deliberate trade-off:

- update failures already fail-tag after **one** attempt (#990), so this is strictly more forgiving;
- `OCR_MAX_RETRIES=0` is the escape hatch for users who prefer the old behavior;
- distinguishing transient from permanent errors would need provider-error classification — out of scope here, happy to iterate if you want it.

Known limitation: the counter is in-memory, so a restart resets it — at worst a persistently failing document gets another `OCR_MAX_RETRIES` attempts per restart. (Documents already fail-tagged are out of the queue for good.)

## Tests

`mockDocumentProcessor` previously could not fail at all — the OCR error path had zero coverage. This PR adds a configurable error (global and per-document) plus `GetDocumentsByTag` support on the existing `recordingClient`, and covers:

- fail tag applied exactly on the configured attempt (rounds 1–2: no PATCH, error surfaces; round 3: one recovery PATCH with `RemoveTags=[auto-OCR tag]`, `SuggestedTags=[FAIL_TAG]`, no loop error),
- successful OCR resets the counter,
- a failed recovery PATCH keeps the counter and is retried next cycle,
- loop shutdown is never counted, while provider-side timeouts (error wrapping `context.DeadlineExceeded` with a live loop context) do count,
- `OCR_MAX_RETRIES=0` keeps the old behavior,
- documents are counted independently (a healthy document never absorbs or resets a broken one's count).

`go build ./... && go test ./...` → 244 tests green; the new tests also pass with `-race`. No frontend changes.

## Docs

README: `FAIL_TAG` description (compose example + env table) extended with the new case, and `OCR_MAX_RETRIES` documented in both places.

## Possible follow-up

The metadata path has the symmetric gap: a persistent `generateDocumentSuggestions` failure also retries forever. The new helper is generic, so extending the same counter approach there is a small change — I left it out to keep this PR focused on the OCR path, but I'm happy to do it as a follow-up if wanted.

## Summary by CodeRabbit

* **New Features**
  * Added `OCR_MAX_RETRIES` to cap consecutive auto-OCR failures per document (default: 3; `0` retries indefinitely).
  * After the cap is reached, the auto-OCR trigger is removed and the configured failure tag is applied; retry behavior is per-document and resets after a successful OCR run.
* **Bug Fixes**
  * Ensure failure streaks aren’t incremented during clean shutdown/cancellation.
* **Documentation**
  * Updated Docker Compose and environment variable docs to describe the new retry and failure-tag behavior.
* **Tests**
  * Added coverage for max-retry fail-tagging, counter reset, per-document isolation, and failure handling edge cases.

### CODE REVIEW

File: background.go [hunk 1]
```diff
@@ -329,9 +369,32 @@ func (app *App) processAutoOcrTagDocuments(ctx context.Context) (int, error) {
...
+					// handled error keep the whole loop in backoff.
+					continue
+				}
+			}
 			errs = append(errs, fmt.Errorf("document %d OCR error: %w", document.ID, err))
 			continue
 		}
+		// OCR itself succeeded — a previous failure streak is over,
+		// whatever the update below does.
+		app.ocrFailures.reset(document.ID)
```
[the remark this answers is not in the document]
Comment (lunetics) [reply background.go:426]: Good catch — Azure DocAI indeed wraps its call in its own `context.WithTimeout` (ocr/azure_provider.go), so matching `context.Canceled`/`DeadlineExceeded` in the returned error would have exempted per-document provider timeouts from the retry limit. Fixed in 8a798aa: the exclusion now checks the poll loop's own `ctx.Err()` instead, and there's a new test asserting that provider-side timeouts do count toward `OCR_MAX_RETRIES`.

**Comment by icereed** [comment]
Vielen Dank! :)
